hyp3_gamma/insar/water_mask.py

```diff
@@ -0,0 +1,146 @@
+    xmin, ymax = upper_left
+    xmax, ymin = lower_right
+    res = trans[1]
+
+    shape_file = f'{mask_location}/GSHHG/GSHHS_f_L1.shp'
+    reproj_shape_file = 'reproj_shape_file.shp'
+    reproject_shapefile(in_tif, shape_file, reproj_shape_file, safe_dir)
+
+    src_ds = ogr.Open(reproj_shape_file)
+    src_lyr = src_ds.GetLayer()
+
+    logging.info("Using xmin, xmax {} {}, ymin, ymax {} {}".format(xmin, xmax, ymin, ymax))
+
+    ncols = int((xmax - xmin) / res + 0.5)
+    nrows = int((ymax - ymin) / res + 0.5)
+
+    logging.info("Creating water body mask of size {} x {} (lxs) using {}".format(nrows, ncols,
+                 reproj_shape_file))
+
+    geotransform = (xmin, res, 0, ymax, 0, -res)
+    dst_ds = gdal.GetDriverByName('GTiff').Create('final_water_mask.tif', ncols, nrows, 1, gdal.GDT_Byte)
+    dst_rb = dst_ds.GetRasterBand(1)
+    dst_rb.Fill(0)
+    dst_ds.SetGeoTransform(geotransform)
+    dst_ds.SetProjection(proj)
+    _ = gdal.RasterizeLayer(dst_ds, [1], src_lyr, burn_values=[mask_value])
+    dst_ds.FlushCache()
+    mask = dst_ds.GetRasterBand(1).ReadAsArray()
+    del dst_ds
+    return mask
+
+
+def apply_water_mask(tiffile, safe_dir, outfile=None, mask=None, maskval=None):
+    """
+
+   Given a tiffile input, fill the masked pixels with NoDataValue of the file.
+
+    """
+    if mask is None:
+        mask = get_water_mask(tiffile, safe_dir, maskval=1)
+
+    if outfile:
+        shutil.copyfile(tiffile, outfile)
+    else:
+        outfile = tiffile
+
+    src_ds = gdal.Open(outfile, gdal.GA_Update)
+    logging.info("Applying water body mask")
+
+    # mask raster
+    for i in range(src_ds.RasterCount):
+        out_band = src_ds.GetRasterBand(i+1)
+        out_data = out_band.ReadAsArray()
+        no_data_value = out_band.GetNoDataValue()
+        out_data[mask == 0] = no_data_value
+        out_band.WriteArray(out_data)
+    # close dataset and flush cache
+    del src_ds
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        prog='mask_water_bodies.py',
+        description='Create and employ a water body mask wherein all water is 0 and land is 1'
+    )
+
+    parser.add_argument('tiffile', help='Name of tif file to mask')
+    parser.add_argument('safedir', help='path of safe_dir')
+    parser.add_argument('outfile', help='Name of output masked file')
+    parser.add_argument('-m', '--mask_value', help='Mask value to apply; default None', type=float,
+                        default=None)
+    args = parser.parse_args()
+
+    logFile = "apply_water_mask_{}_log.txt".format(os.getpid())
+    logging.basicConfig(filename=logFile, format='%(asctime)s - %(levelname)s - %(message)s',
+                        datefmt='%m/%d/%Y %I:%M:%S %p', level=logging.DEBUG)
+    logging.getLogger().addHandler(logging.StreamHandler())
+    logging.info("Starting run")
+
+    apply_water_mask(args.tiffile, args.safedir, outfile=args.outfile, maskval=args.mask_value)
+
+    if __name__ == '__main__':
+        main()
```

Comment:
All this implies you want to provide a `mask_water_bodies.py` script -- if so, you need to add `'mask_water_bodies.py = hyp3_gamma.insar.water_mask:main',` to the [entrypoints in setup.py](https://github.com/ASFHyP3/hyp3-gamma/blob/include_water_mask/setup.py#L59-L66) and a test for it in [tests/test_entrypoints.py](https://github.com/ASFHyP3/hyp3-gamma/blob/include_water_mask/tests/test_entrypoints.py)